provider/ec2/environ.go

```diff
@@ -180,10 +181,12 @@ func (e *environ) ConstraintsValidator(ctx context.ProviderCallContext) (constra
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	sort.Sort(instances.ByName(instanceTypes))
```

Comment:
A test for this might be worth while.

Reply:
:+1:

Reply:
@SimonRichardson  added a test for the instance-type place itself or did you mean in the provider package itself?

Reply:
I meant testing the output from that, along with the sort.